### ISSUE TITLE
Just curious, are the changes for Surface being pushed to main Linux kernel development so we can hope to see in common Linux distros at some point?

### ISSUE THREAD

**Issue by LarryMarrs-Utah** [jakeday/linux-surface#581]

**Comment by qzed** [comment]
I'm trying to get the changes I made upstream. Button patches for the newer Surface devices should be in 5.4 (are already in 5.4-rc1).

The Surface ACPI stuff will probably still take a while, but I'm currently working on getting the patches required for it upstream (see https://lkml.org/lkml/2019/9/24/481 and the previous revision of that patch https://lkml.org/lkml/2019/9/19/755). I'm also waiting on feedback for another issue which needs to be resolved for that. That's probably a bit more tricky as it results from Microsoft not following the ACPI specification (which Linux follows).

Besides that, we're always trying to improve the current patches (currently focusing a lot on IPTS) so that they can hopefully at some point be considered for upstreaming.
